src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHelpers.cs

```diff
@@ -0,0 +1,311 @@
+            byte[] buffer = ArrayPool<byte>.Shared.Rent(minimumLength: MaxBufferLength);
+            while (bytesToCopy > 0)
+            {
+                int currentLengthToRead = (int)Math.Min(MaxBufferLength, bytesToCopy);
+                if (origin.Read(buffer.AsSpan(0, currentLengthToRead)) != currentLengthToRead)
+                {
+                    throw new EndOfStreamException();
+                }
+                destination.Write(buffer.AsSpan(0, currentLengthToRead));
+                bytesToCopy -= currentLengthToRead;
+            }
+            ArrayPool<byte>.Shared.Return(buffer);
+        }
+
+        // Returns the number of bytes until the next multiple of the record size.
+        internal static int CalculatePadding(long size)
+        {
+            long ceilingMultipleOfRecordSize = ((RecordSize - 1) | (size - 1)) + 1;
+            int padding = (int)(ceilingMultipleOfRecordSize - size);
+            return padding;
+        }
+
+        // Returns the specified 8-base number as a 10-base number.
+        internal static int ConvertDecimalToOctal(int value)
+        {
+            int multiplier = 1;
+            int accum = value;
+            int actual = 0;
+            while (accum != 0)
+            {
+                actual += (accum % 8) * multiplier;
+                accum /= 8;
+                multiplier *= 10;
+            }
+            return actual;
+        }
+
+        // Returns the specified 10-base number as an 8-base number.
+        internal static long ConvertDecimalToOctal(long value)
+        {
+            long multiplier = 1;
+            long accum = value;
+            long actual = 0;
+            while (accum != 0)
+            {
+                actual += (accum % 8) * multiplier;
+                accum /= 8;
+                multiplier *= 10;
+            }
+            return actual;
+        }
+
+        // Returns true if all the bytes in the specified array are nulls, false otherwise.
+        internal static bool IsAllNullBytes(Span<byte> buffer)
+        {
+            for (int i = 0; i < buffer.Length; i++)
+            {
+                if (buffer[i] != 0)
+                {
+                    return false;
+                }
+            }
+            return true;
+        }
+
+        // BitConverter.GetBytes returns a byte array with more cells than necessary, while Encoding.*.GetBytes
+        // returns an array of the exact number of cells required to store the converted number value.
+        // int overload.
+        internal static byte[] GetAsciiBytes(int number) => Encoding.ASCII.GetBytes(number.ToString());
+
+        // BitConverter.GetBytes returns a byte array with more cells than necessary, while Encoding.*.GetBytes
+        // returns an array of the exact number of cells required to store the converted number value.
+        // long overload.
+        internal static byte[] GetAsciiBytes(long number) => Encoding.ASCII.GetBytes(number.ToString());
```

Comment:
If after my previous comments these are still needed, they can be implemented more efficiently (e.g. not allocating the intermediate string).  Also, are these numbers always non-negative?  If they can be negative, these should be passed InvariantCulture.